homework3/semi-uber/UI/src/app/user-page/user-page.component.ts

```diff
@@ -1,15 +1,39 @@
-import { Component, OnInit } from '@angular/core';
+import { UserShipper } from "./../interfaces/user-shipper.inerfase copy";
+import { UserDriver } from "./../interfaces/user-driver.inerfase";
+import { Component, OnInit } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { environment as env } from "../../environments/environment";
+import { ActivatedRoute } from "@angular/router";
 
 @Component({
-  selector: 'app-user-page',
-  templateUrl: './user-page.component.html',
-  styleUrls: ['./user-page.component.scss']
+  selector: "app-user-page",
+  templateUrl: "./user-page.component.html",
+  styleUrls: ["./user-page.component.scss"]
 })
 export class UserPageComponent implements OnInit {
-
-  constructor() { }
+  user: any = {};
+  constructor(private http: HttpClient, private route: ActivatedRoute) {}
 
   ngOnInit(): void {
+    this.getUserData();
   }
 
+  getUserData() {
+    const token = localStorage.getItem("JWT");
+    const id = this.route.snapshot.paramMap.get("id");
+    this.http
+      .get(`${env.baseURL}/user/${id}`, {
```

Comment:
don't forget to move all http requests to a separate service during FE refactoring